app/src/main/cpp/HandMeshRenderer.cpp

```diff
@@ -0,0 +1,634 @@
+/* -*- Mode: C++; tab-width: 20; indent-tabs-mode: nil; c-basic-offset: 2 -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "Controller.h"
+#include "DeviceUtils.h"
+#include "HandMeshRenderer.h"
+#include "tiny_gltf.h"
+
+#include "vrb/Camera.h"
+#include "vrb/Color.h"
+#include "vrb/ConcreteClass.h"
+#include "vrb/GLError.h"
+#include "vrb/ProgramFactory.h"
+#include "vrb/RenderState.h"
+#include "vrb/ShaderUtil.h"
+#include "vrb/Toggle.h"
+#include "vrb/gl.h"
+
+#define XR_EXT_HAND_TRACKING_NUM_JOINTS 26
+
+namespace crow {
+
+// HandMeshRendererSpheres
+
+struct HandMeshSpheres {
+    vrb::TogglePtr toggle;
+    std::vector<vrb::TransformPtr> sphereTransforms;
+};
+
+struct HandMeshRendererSpheres::State {
```

Comment:
Is the struct really needed? Are we going to add more fields to the struct? If not I guess we can get rid of it.

Reply:
A State struct is needed by the vrb::CreationContext pattern. I find this construction cleaner that polluting the header file.

Many classes in Wolvic use this pattern and I think there is a good reason for it: it keeps implementation details out of header files. This causes that changes in an implementation would not trigger a rebuild of the whole world because a new class variable was needed by the implementation.

Do you feel strongly about removing this Struct and move everything to the header? I would also have to stop using the vrb::CreationContext pattern.

Reply:
So the `vrb::ConcreteClass` is just there to do a simplified leak detection. I can understand using it for things like nodes in the scenegraph. However for the case of strategies I don't think it will benefit us in any way. We're creating just one instance which is stored in an unique_ptr that will be freed once the owner of the pointer is freed. So there are not really benefits on using it from my POV.